mobile/Features/CheckIn/AddColor.jsx

```diff
@@ -0,0 +1,89 @@
+import { useState } from 'react';
+import {
+  Text, View, Pressable, Image,
+} from 'react-native';
+import { useRoute } from '@react-navigation/native';
+import ProgressBar from 'react-native-progress/Bar';
+import PropTypes from 'prop-types';
+import tempColor from '../../assets/colorSquare.png';
+import styles from './AddColorStyles';
+
+function AddColor({ navigation }) {
+  // get parameters from route (setAddedMoods, addedMoods, moodPassedIn)
+  const route = useRoute();
+  const setAddedMoods = route.params?.setAddedMoods;
+  const addedMoods = route.params?.addedMoods;
+  const moodPassedIn = route.params?.mood;
+
+  // set progress
+  const numPages = route.params?.numPages;
+  const progress = 1 / numPages;
+
+  // array of rows
+  // each row contains color, image pairs
+  const colorToPicture = [
+    [['red', tempColor], ['blue', tempColor], ['47', tempColor]],
+    [['dog', tempColor], ['Nicole', tempColor], ['no', tempColor]],
+    [['skeeeeyee', tempColor], ['help', tempColor], ['how', tempColor]],
+  ];
+
+  // state to keep track of which color the user chose
+  const [colorChosen, setColorChosen] = useState('');
+
+  // update the addedMoods array from the original mood screen with the mood/color the user chose
+  // also navigate back to the original mood screen
+  const continueButton = () => {
+    const newAddedMoods = [...addedMoods];
+    newAddedMoods.push([moodPassedIn, colorChosen]);
+    setAddedMoods(newAddedMoods);
+    navigation.navigate('Mood', { numPages });
+  };
+
+  return (
+    <View style={styles.container}>
+      <ProgressBar progress={progress} width={200} style={{ top: '5%' }} />
+      <View style={styles.heading}>
+        <Text>
+          now, select a color to represent this mood!
+        </Text>
+      </View>
+      <View style={styles.content}>
+        {colorToPicture.map((pairs) => (
```

Comment:
great work with the .map! 
I think the code can be made more clean with a nested map though, and i think it would be good practice!
you got this :)